cmake/common/buildspec_common.cmake

```diff
@@ -0,0 +1,215 @@
+# Common build dependencies module
+
+# cmake-format: off
+# cmake-lint: disable=C0103
+# cmake-lint: disable=E1126
+# cmake-lint: disable=R0912
+# cmake-lint: disable=R0915
+# cmake-format: on
+
+include_guard(GLOBAL)
+
+# _check_deps_version: Checks for obs-deps VERSION file in prefix paths
+function(_check_deps_version version)
+  # cmake-format: off
+  set(found FALSE PARENT_SCOPE)
+  # cmake-format: on
+
+  foreach(path IN LISTS CMAKE_PREFIX_PATH)
+    if(EXISTS "${path}/share/obs-deps/VERSION")
+      if(dependency STREQUAL qt6 AND NOT EXISTS "${path}/lib/cmake/Qt6/Qt6Config.cmake")
+        # cmake-format: off
+        set(found FALSE PARENT_SCOPE)
+        # cmake-format: on
+        continue()
+      endif()
+
+      file(READ "${path}/share/obs-deps/VERSION" _check_version)
+      string(REPLACE "\n" "" _check_version "${_check_version}")
+      string(REPLACE "-" "." _check_version "${_check_version}")
+      string(REPLACE "-" "." version "${version}")
+
+      if(_check_version VERSION_EQUAL version)
+        # cmake-format: off
+        set(found TRUE PARENT_SCOPE)
+        # cmake-format: on
+        break()
+      elseif(_check_version VERSION_LESS version)
+        message(AUTHOR_WARNING "Outdated ${label} version detected in ${path}: \n"
```

Comment:
"Older" instead of "Outdated"? I feel less strongly about this than the "Future" line below.